### PR TITLE
refactor(test): replace mockr to testthat's mock functions

### PR DESCRIPTION
**Pull request by eitsupi**
testthat now have the new mocking tools (r-lib/testthat#1739).

So, the mockr package is no longer needed.

### CODE REVIEW

File: tests/testthat/_snaps/rust-siterp.md [hunk 1]
```diff
@@ -1,13 +0,0 @@
-# `cargo` or `rustup` are not found
```
Comment (Ilia-Kosenkov) [inline tests/testthat/_snaps/rust-siterp.md:1]: Why this snapshot is now removed?

Comment (eitsupi) [reply tests/testthat/_snaps/rust-siterp.md:1]: Please pay attention to the filename. This file seems to have been added by typo and is not really used.

Comment (Ilia-Kosenkov) [reply tests/testthat/_snaps/rust-siterp.md:1]: Oh, ok, I see, I now remember I messed up with test naming when I was developing `rust_sitrep()` and forgot to remove the snapshot!

**Comment by eitsupi** [comment]
Sorry, I forgot to bump the version of testthat.
(These mock functions were added on 3.1.7)

**Comment by Ilia-Kosenkov** [comment]
I forgot, can you merge it yourself after approval?

**Comment by eitsupi** [comment]
I do not have merge rights to this repository, so cannot merge.

**Pull request merged**